datalad/interface/add_archive_content.py

```diff
@@ -323,6 +323,17 @@ def __call__(archive, annex=None,
             stats = ActivityStats()
 
             for extracted_file in earchive.get_extracted_files():
+                # continue to next iteration if extracted_file in excluded
+                if exclude:
```

Comment:
ok, I can live with this solution I think ;)  just move it after `stats.files += 1` line